app/src/main/java/fragments/TempFeedFragment.java

```diff
@@ -62,6 +81,16 @@ public class TempFeedFragment extends Fragment implements AdapterView.OnItemSele
     public PostAdapter adapter;
     SwipeRefreshLayout swipeContainer;
 
+    // Weather data
+    private String description;
+    private String temperature;
+    // sunset time of local timezone
+    private String sunsetTime;
+    long myTimeZone = -25200;
+
+    private final String url = "https://api.openweathermap.org/data/2.5/weather";
+    private final String api_key = BuildConfig.WEATHER_KEY;
+
     public TempFeedFragment() {
         // Required empty public constructor
     }
```

Comment:
informational: i generally create a public static newInstance() method in my fragments which returns an object of your fragment. and then i make the constructor for the fragment private. this is mostly important if i have any required parameters for my fragments and want to ensure anyone creating the fragment includes those parameters.

in your case though, you have no parameters so no big deal


but for example

public static TempFeedFragment newInstance(final P parameter){
    Bundle args = new Bundle();
    args.putString("parameter", parameter);
    TempFeedFragment f = new TempFeedFragment();
    f.setArguments(args);
    return f;
}

private TempFeedFragment() {} // disallow construction of your fragment outside of this class